components/styled/table/src/molecules/table-core.tsx

```diff
@@ -0,0 +1,137 @@
+import styled from '@emotion/styled'
+import { Icon, IconButton } from '@ltht-react/icon'
+import { TEXT_COLOURS, TRANSLUCENT_BRIGHT_BLUE_TABLE } from '@ltht-react/styles'
+import { flexRender, Table } from '@tanstack/react-table'
+import { Dispatch, SetStateAction } from 'react'
+import { CellProps } from './table-cell'
+import { StandardButtonStyle, StyledTableData, StyledTableHeader } from './table-styles'
+
+const DefaultTableOptions = {
+  showExpanderColumn: false,
+  pageSize: 10,
+  enablePagination: false,
+  serverSidePagination: false,
+  hidePaginationControls: false,
+  hidePerPageOptions: false,
+}
+
+const DefaultPerPageOptions = [10, 20, 30, 40, 50]
+
+const buildTableHead = (table: Table<DataEntity>) => (
+  <thead>
+    {table.getHeaderGroups().map((headerGroup) => (
+      <tr key={headerGroup.id} role="row">
+        {headerGroup.headers.map((header) =>
+          header.column.id === 'expander' ? (
+            <StyledTableHeader key={header.id} colSpan={header.colSpan} role="columnheader">
+              {header.isPlaceholder ? null : flexRender(header.column.columnDef.header, header.getContext())}
+            </StyledTableHeader>
+          ) : (
+            <StyledTableHeader
+              role="columnheader"
+              key={header.id}
+              colSpan={header.colSpan}
+              {...{
+                style: {
+                  cursor: header.column.getCanSort() ? 'pointer' : '',
+                },
+                onClick: header.column.getToggleSortingHandler(),
+              }}
+            >
+              {header.isPlaceholder ? null : flexRender(header.column.columnDef.header, header.getContext())}
+            </StyledTableHeader>
+          )
+        )}
+      </tr>
+    ))}
+  </thead>
+)
+
+const buildTableBody = (table: Table<DataEntity>) => (
+  <tbody>
+    {table.getRowModel().rows.map((row) => (
+      <tr key={row.id} role="row">
+        {row.getVisibleCells().map((cell, cellIdx) => (
+          <StyledTableData
+            role="cell"
+            key={cell.id}
+            style={{
+              background: cellIdx % 2 === 1 ? 'white' : TRANSLUCENT_BRIGHT_BLUE_TABLE,
+              textAlign: 'center',
+            }}
+          >
+            {flexRender(cell.column.columnDef.cell, cell.getContext())}
+          </StyledTableData>
+        ))}
+      </tr>
+    ))}
+  </tbody>
+)
+
+const StyledStandardButton = styled(IconButton)`
+  ${StandardButtonStyle}
+`
+const displayErrorMessage = (message: string, setState: Dispatch<SetStateAction<boolean>>) => (
+  <>
+    <div style={{ padding: 10, borderRadius: 10 }}>
+      <Icon type="exclamation" size="medium" style={{ color: TEXT_COLOURS.DANGER, marginRight: 5 }} />
+      {message}
+    </div>
+    <div style={{ display: 'flex', justifyContent: 'center' }}>
+      <StyledStandardButton
+        iconProps={{ type: 'spinner', size: 'small', animate: false }}
```

Comment:
I'm getting a couple TypeScript errors like this after pulling it down, not sure if it's just me but thought I'd let you know 
![image](https://user-images.githubusercontent.com/113031996/202211039-9fa5b3ee-e8cb-47ae-8b20-bf2924d65f77.png)

Reply:
You probably need to run `yarn clean` => `yarn install` => `yarn build` to update your lib folders with the new types :)

Reply:
Thanks, I'll do that now